### PR TITLE
add configurable max age to tablecache

### PR DESCRIPTION
**Pull request by jlerche**
Took a stab at #138, not sure if this is in line with what you had in mind.

### CODE REVIEW

File: tabcache/tablecache.go [hunk 1]
```diff
@@ -73,6 +85,13 @@ func (tc *Cache) AddSSTable(tableID sst.SSTableID, table *sst.SSTable) error {
 	return nil
 }
 
+func (tc *Cache) AddSSTableWithMaxAge(tableID sst.SSTableID, table *sst.SSTable) error {
```
Comment (jlerche) [inline tabcache/tablecache.go:88]: Might make more sense to call this `...WithDefaultMaxAge`

**Comment by purplefox** [comment]
Hey @jlerche I am just trying to figure out whether this task makes sense.
I made the assumption when creating the task that when executing a query that pulls in old tables, they are added to the cache, but checking the code this doesn't seem to be the case - tables are only added to the cache when they're built before pushing or when they're built during compaction. So the original perceived issue is a non issue.
However, I suppose when compacting, we don't want to add tables which are created from other tables that were originally pushed a longer time ago. So we could keep your changes, and when we add the table to the cache in the compaction worker, add it with age given by MaxAddedTime.
Also, we should make the max age much smaller, the purpose of the cache is to enable quick retrieval of very recent tables, e.g. for when getting last value in an aggregation, so we should limit max age to 5 mins or so.

**Comment by jlerche** [comment]
> Hey @jlerche I am just trying to figure out whether this task makes sense. I made the assumption when creating the task that when executing a query that pulls in old tables, they are added to the cache, but checking the code this doesn't seem to be the case - tables are only added to the cache when they're built before pushing or when they're built during compaction. So the original perceived issue is a non issue. However, I suppose when compacting, we don't want to add tables which are created from other tables that were originally pushed a longer time ago. So we could keep your changes, and when we add the table to the cache in the compaction worker, add it with age given by MaxAddedTime. Also, we should make the max age much smaller, the purpose of the cache is to enable quick retrieval of very recent tables, e.g. for when getting last value in an aggregation, so we should limit max age to 5 mins or so.

Gotcha that makes sense, I'll go ahead and revert the change to `storage.go` and use the with max age in the compaction worker.

**Comment by purplefox** [comment]
Table cache test seems to be failing with these changes.

**Comment by jlerche** [comment]
> Table cache test seems to be failing with these changes.

New commit should fix it, although I'm not sure why the new change caused the test to fail, or why the test was working before the change. Also updated the `TestParseConfigWithComments` test

**Comment by purplefox** [comment]
> Also updated the TestParseConfigWithComments test

Great!

Could you also update `conf/conf_test.go` ?

**Comment by jlerche** [comment]
> > Also updated the TestParseConfigWithComments test
> 
> Great!
> 
> Could you also update `conf/conf_test.go` ?

Yep :rocket:

**Comment by purplefox** [comment]
Thank you

**Pull request merged**